examples/step-69/step-69.cc

```diff
@@ -0,0 +1,2926 @@
+  // actual final values for the parameters, this would be sort of
+  // meaningless an unsafe in general (we would like to have mechanisms to
+  // check the consistency of the input parameters). Instead of defining
+  // another <code>setup()</code> method to be called (by-hand) after the
+  // call to <code> ParameterAcceptor::initialize() </code> we provide an
+  // "implementation" for the class member
+  // <code>parse_parameters_call_back</code> which is automatically called when
+  // invoking <code> ParameterAcceptor::initialize() </code> for every class
+  // that inherits from ParameterAceptor.
+
+  template <int dim>
+  class InitialValues : public ParameterAcceptor
+  {
+  public:
+    using rank1_type = typename ProblemDescription<dim>::rank1_type;
+
+    InitialValues(const std::string &subsection = "InitialValues");
+
+    std::function<rank1_type(const Point<dim> &point, double t)> initial_state;
+
+  private:
+    // We declare a private callback function that will be wired up to the
+    // ParameterAcceptor::parse_parameters_call_back signal.
+    void parse_parameters_callback();
+
+    Tensor<1, dim> initial_direction;
+    Tensor<1, 3>   initial_1d_state;
+  };
+
+  // @sect4{The <code>TimeStep</code> class}
+  //
+  // With the <code>OfflineData</code> and <code>ProblemDescription</code>
+  // classes at hand we can now implement the explicit time-stepping scheme
+  // that was introduced in the discussion above. The main method of the
+  // <code>TimeStep</code> class is <code>step(vector_type &U, double
+  // t)</code> that takes a reference to a state vector <code>U</code> and
+  // a time point <code>t</code> (as input arguments) computes the updated
+  // solution, stores it in the vector <code>temp</code>, swaps its contents
+  // with the vector <code>U</code>, and returns the chosen step-size
+  // $\tau$.
+  //
+  // The other important method is <code>prepare()</code> which primarily
+  // sets the proper partition and sparsity pattern for the temporary
+  // vector <code>temp</code> and the matrix <code>dij_matrix</code>
+  // respectively.
+
+  template <int dim>
+  class TimeStep : public ParameterAcceptor
+  {
+  public:
+    static constexpr unsigned int problem_dimension =
+      ProblemDescription<dim>::problem_dimension;
+
+    using rank1_type = typename ProblemDescription<dim>::rank1_type;
+    using rank2_type = typename ProblemDescription<dim>::rank2_type;
+
+    typedef std::array<LinearAlgebra::distributed::Vector<double>,
+                       problem_dimension>
+      vector_type;
+
+    TimeStep(const MPI_Comm &          mpi_communicator,
+             TimerOutput &             computing_timer,
+             const OfflineData<dim> &  offline_data,
+             const InitialValues<dim> &initial_values,
+             const std::string &       subsection = "TimeStep");
+
+    void prepare();
+
+    double step(vector_type &U, double t);
```

Comment:
maybe `do_one_timestep`? "step" can be both a noun and a verb. I think you want to express an action "verb", but it's not quite clear from the name alone.

Reply:
changed.